internal/engine/upper.go

```diff
@@ -494,14 +494,19 @@ func handleConfigsReloaded(
 		newDefOrder[i] = m.ManifestName()
 
 		configFilesThatChanged := state.LastTiltfileBuild.Edits
-		if !m.Equal(mt.Manifest) {
+		old := mt.Manifest
+		if !m.Equal(old) {
+			old := mt.Manifest
 			mt.Manifest = m
```

Comment:
Moving this outside the `if` would reduce the impact of similar bugs in the future, though with the downside that the remaining impact would be harder to notice.

Reply:
😬 imma do it!